### PR TITLE
well_stocked

### PR DESCRIPTION
**Pull request by jaadams5**
See https://kolmafia.us/threads/well_stocked-ash-function.26815/

boolean well_stocked(item, quantity, price)

Answers the question "can I get quantity of item spending no more than price for the last item?"

I'm not convinced I am handling and checking parameters in the best way and I have done nothing about testing.

Test script requires a logged in character so....

Spotless decided it wanted to rearrange some imports or my IDE wildcarded them and spotless doesn't care either way.

### CODE REVIEW

File: src/net/sourceforge/kolmafia/textui/RuntimeLibrary.java [hunk 1]
```diff
@@ -5066,6 +5038,41 @@ public static Value mall_prices(
...
+        int storePrice = pr.getPrice();
+        if (storePrice > checkPrice) return new Value(available >= checkQuant);
+        // get available
+        int canGet = Math.min(pr.getLimit(), pr.getQuantity());
+        available += canGet;
+        if (available >= checkQuant) return new Value(true);
+      }
+    }
+    // if we get here we have failed to find enough stock
+    return new Value(false);
```
Comment (gausie) [inline src/net/sourceforge/kolmafia/textui/RuntimeLibrary.java:5073]:
```suggestion
    return DataTypes.FALSE_VALUE;
```

File: src/net/sourceforge/kolmafia/textui/RuntimeLibrary.java [hunk 2]
```diff
@@ -5066,6 +5038,41 @@ public static Value mall_prices(
...
+    for (PurchaseRequest pr : results) {
+      // only interested in mall
+      if (pr instanceof MallPurchaseRequest) {
+        // get price and bail if higher
+        int storePrice = pr.getPrice();
+        if (storePrice > checkPrice) return new Value(available >= checkQuant);
+        // get available
+        int canGet = Math.min(pr.getLimit(), pr.getQuantity());
+        available += canGet;
+        if (available >= checkQuant) return new Value(true);
```
Comment (gausie) [inline src/net/sourceforge/kolmafia/textui/RuntimeLibrary.java:5069]:
```suggestion
        if (available >= checkQuant) return DataTypes.TRUE_VALUE;
```

File: src/net/sourceforge/kolmafia/textui/RuntimeLibrary.java [hunk 3]
```diff
@@ -5066,6 +5038,41 @@ public static Value mall_prices(
...
+    // extract parameters
+    String item = itemName.toString();
+    int itemID = ItemDatabase.getItemId(item);
+    long checkQuant = quantity.intValue();
+    long checkPrice = price.intValue();
+    // check parameters and return false if problems
+    if (item == null) return new Value(false);
+    if (itemID < 1) return new Value(false);
+    if (checkQuant < 6) return new Value(false);
+    if (checkPrice < (2 * ItemDatabase.getPriceById(itemID))) return new Value(false);
```
Comment (gausie) [inline src/net/sourceforge/kolmafia/textui/RuntimeLibrary.java:5052]:
```suggestion
    if (item == null) return DataTypes.FALSE_VALUE;
    if (itemID < 1) return DataTypes.FALSE_VALUE;
    if (checkQuant < 6) return DataTypes.FALSE_VALUE;
    if (checkPrice < (2 * ItemDatabase.getPriceById(itemID))) return DataTypes.FALSE_VALUE;
```

File: src/net/sourceforge/kolmafia/textui/RuntimeLibrary.java (hunk 2 above)
Comment (jaadams5) [reply src/net/sourceforge/kolmafia/textui/RuntimeLibrary.java:5069]: Again, Good catch.  Thanks.

File: src/net/sourceforge/kolmafia/textui/RuntimeLibrary.java (hunk 3 above)
Comment (jaadams5) [reply src/net/sourceforge/kolmafia/textui/RuntimeLibrary.java:5052]: Fixing now.

**Pull request merged**